### PR TITLE
feat: add locale prop to Settings

### LINKED ISSUE #2157: [xy] Connect multi-time axis to Kibana Locale
The current implementation of the multi-layer time axis uses a default English locale.
The locale should be passed to the chart and used here.

### LINKED ISSUE #2158: Remove specified locale translations for multi-layer time axis
**Describe the issue**
The specified set of locale translation in https://github.com/elastic/elastic-charts/blob/4070da0c1f152009cd3a62abf314f6cb9f711845/packages/charts/src/chart_types/xy_chart/axes/timeslip/locale_translations.ts is used to render the size of the current aggregation in the multilayer timesplit example.
This is not used anywhere else and probably also wrongly represents some locale definitions, like the Japanese coded as `ja-JA` but should be `ja-JP`.

### PR DESCRIPTION
**Pull request by markov00**
## Summary
This PR adds the `locale` prop to the `Settings` component.
This Unicode Locale Identifier is used across the entire charting library whenever a localized comparison or calculation is needed. For example in the multi-layer time axis or in some categorical/numerical comparison.

## Details

I've also removed the manual translation used by the timeslip component because they where wrong and unused.

## Issues

fix #2158 
fix #2157

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [x] New public API exports have been added to `packages/charts/src/index.ts`
- [x] Unit tests have been added or updated to match the most common scenarios
- [x] The proper documentation and/or storybook story has been added or updated

### CODE REVIEW

**Review by nickofthyme — APPROVED** [review]
LGTM.

There's a lot of passing around of `locale` I wonder if we could leverage a global context to access this easier like with `useLocale` hook. No for this PR but something to consider in the future.

File: packages/charts/src/chart_types/timeslip/timeslip/config.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export interface TimeslipConfig extends TimeslipTheme, RasterConfig {
 /** @internal */
 export const rasterConfig: RasterConfig = {
   minimumTickPixelDistance: MINIMUM_TICK_PIXEL_DISTANCE,
-  locale: DEFAULT_LOCALE,
+  locale: 'en-US',
```
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/chart_types/timeslip/timeslip/config.ts:92]: Actually doesn't change much in the timeslip, is just a prototype component actually, but you are right, is different

**Comment by markov00** [comment]
> LGTM.
> 
> There's a lot of passing around of `locale` I wonder if we could leverage a global context to access this easier like with `useLocale` hook. No for this PR but something to consider in the future.

True, but I prefer to not be bonded to React hooks too much, it is also a bit difficult to use hooks here because most usage of the locale are within selectors/utils function rather then react environment.

File: packages/charts/src/chart_types/xy_chart/crosshair/crosshair_utils.linear_snap.test.ts [hunk 2]
```diff
@@ -95,34 +95,39 @@ describe('Crosshair utils linear scale', () => {
     barSeries,
     getScaleConfigsFromSpecs([], barSeries, MockGlobalSpec.settings()),
     [],
+    { locale: 'en' },
```
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/crosshair/crosshair_utils.linear_snap.test.ts]: Updated to en-US everywhere https://github.com/elastic/elastic-charts/pull/2164/commits/4a60938759fd9e37f4120885938eabb6d8781087

File: packages/charts/src/utils/text/wrap.ts [hunk 3]
```diff
@@ -71,7 +71,7 @@ export function wrapText(
   return lines;
 }
 
-function textSegmenter(locale: string[]): (text: string) => { segment: string; index: number; isWordLike?: boolean }[] {
+function textSegmenter(locale: string): (text: string) => { segment: string; index: number; isWordLike?: boolean }[] {
```
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/utils/text/wrap.ts:74]: created an issue here https://github.com/elastic/elastic-charts/issues/2170

**Pull request merged**